opr/primer.py

```diff
@@ -274,10 +305,13 @@ def melting_temperature(self, method=MeltingTemperature.BASIC):
         :type method: MeltingTemperature
         :return: approximated melting temperature
         """
-        if self._melting_temperature[method] is not None:
+        if method not in self._computed["melting_temperature"]:
+            raise OPRBaseError(PRIMER_INVALID_MELTING_TEMPERATURE_METHOD_ERROR)
```

Comment:
This error and `raise NotImplementedError(PRIMER_MELTING_TEMPERATURE_NOT_IMPLEMENTED_ERROR)` are very similar; please properly merge them. I suggest raising only `NotImplementedError`.